addons/mail/models/mail_message.py

```diff
@@ -889,7 +875,7 @@ def _message_format(self, fnames, format_reply=True):
             vals.update({
                 'notifications': message_sudo.notification_ids._filtered_for_web_client()._notification_format(),
                 'attachment_ids': message_sudo.attachment_ids._attachment_format(),
-                'tracking_value_ids': tracking_value_ids,
+                'tracking_value_ids': message_sudo.tracking_value_ids.sudo(self.env.su)._tracking_value_format(),
```

Comment:
Too much sudo and su ^^

Reply:
This is just a move of existing code, there PR is making no change in this regard.

`self.env.is_superuser() or self.user_has_groups(groups)` inside the method has to be done with the `self.env` like before (possibly sudo, possibly not sudo), and not with the message sudo env (incorrectly always sudo in that particular case).

Reply:
Ah ok, that's a bit out of the blue. But as the python method is called only once, your code is more complicated with no added value, unless I misunderstood something ? It seems you could keep the old code as it is.

Reply:
Side quest: maybe filter-out tracking values first, then call the format method on them, like
```
allowed_tracking_ids = ...  # filter as before, just remove the inline dict
vals['tracking_value_ids'] = allowed_tracking_ids._tracking_value_format()

Reply:
Good idea to split filtering and formatting, this should actually be done even if we didn't have those sudo issue.